### DISCUSSION TITLE
OnDelete and Views without introspecting

### DISCUSSION THREAD

**Discussion by euberdeveloper** [prisma/prisma#5577] [Q&A] [answered]
Hello,

I am developing a project and I want to use Prisma. I know that things such as **onDelete: CASCADE** are available on **Prisma 1** but not in **Prisma 2** and that they are **planned** to be implemented.

I also saw that you can use **onDelete: CASCADE** and **views** by adding them **manually in the db** and use an **introspect** of the schema.

What if I do not want to introspect? **What would happen if I created a schema, without cascade and views, modeled my db through that prisma schema, updated my db adding a view or some constraints/delete-cascade, add the view as a prisma schema model and eventually syncronize the prisma model and the db again?**

**Comment by ryands17** [comment, marked as answer]
Hey @euberdeveloper 👋 

> know that things such as onDelete: CASCADE are available on Prisma 1 but not in Prisma 2 and that they are planned to be implemented.

For this, I would suggest a better workaround for the moment i.e. using this [plugin](https://paljs.com/plugins/delete) that makes it easy to apply cascade options on your Prisma schema.

> What would happen if I created a schema, without cascade and views, modeled my db through that prisma schema, updated my db adding a view or some constraints/delete-cascade, add the view as a prisma schema model and eventually syncronize the prisma model and the db again?

This is possible and you don't need to add the views separately, you can directly add them via [Prisma Migrate](https://www.prisma.io/docs/concepts/components/prisma-migrate) where you can directly create your view in an `.sql` file and that will be added to your database by Prisma when your run Migrate.

Then you will need to add the view to the schema manually then run `prisma generate` and you will be able to query from the view.

Let me know if that helps :)

**Comment by euberdeveloper** [comment]
Yes, I think that it helped.

The first option is interesting, but it solves only the on delete cascade, so I think that the most flexible option will be to use migration

**Reply by ryands17** [reply]
> but it solves only the on delete cascade

The plugin supports `set null` as well.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
